doc/source/admin/special_topics/vault.md

```diff
@@ -0,0 +1,120 @@
+
+| Backend     | Description                                                                                                                                                                                                                                                                                                                                   |
+|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| hashicorp   | Hashicorp Vault is a secrets and encryption management system. https://www.vaultproject.io/                                                                                                                                                                                                                                                   |
+| custos      | Custos is an NSF-funded project, backed by open source software that provides science gateways such as Galaxy with single sign-on, group management, and management of secrets such as access keys and OAuth2 access tokens. Custos secrets management is backed by Hashicorp's vault, but provides a convenient, always-on ReST API service. |
+| database    | The database backend stores secrets in an encrypted table in the Galaxy database itself. It is a convenient way to get started with a vault, and while it supports basic key rotation, we recommend using one of the above 2 options in production.                                                                                           |
+
+## Configuring Galaxy
+
+The first step to using a vault is to configure the `vault_config_file` setting in `galaxy.yml`. 
+
+```yaml
+galaxy:
+  # Vault config file
+  # The value of this option will be resolved with respect to
+  # <config_dir>.
+  vault_config_file: vault_conf.yml
+```
+
+## Configuring vault_conf.yml
+
+The vault_conf.yml file itself has two basic fields:
+```yaml
+type: hashicorp  # required
+path_prefix: /galaxy  # optional
+...
+```
+
+The `type` must be a valid backend type. The `path_prefix` property indicates the root path under which to store all vault keys. If multiple Galaxy instances are using the same Vault, a prefix can be used to identify the galaxy instance uniquely for example.
+If no path_prefix is provided, the prefix defaults to `/galaxy`.
+
+## Vault configuration for Hashicorp Vault
+
+```yaml
+type: hashicorp
+path_prefix: /my_galaxy_instance
+vault_address: http://localhost:8200
+vault_token: galaxy_test_token
+```
+
+## Vault configuration for Custos
+
+```yaml
+type: custos
+custos_host: service.staging.usecustos.org
+custos_port: 30170
+custos_client_id: custos-jeREDACTEDye-10000001
+custos_client_sec: OGREDACTEDBSUDHn
+```
+
```

Comment:
Obtaining the Custos client id and client secret requires first registering your Galaxy instance with Custos. Visit [usecustos.org](http://usecustos.org/) for more information.